uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -1,32 +1,61 @@
 import asyncio
 import http
 import logging
-from typing import Callable
+import sys
+from typing import Any, Callable, List, Optional, Sequence, Tuple, Union, cast
 from urllib.parse import unquote
 
 import websockets
+from asgiref.typing import (
+    ASGISendEvent,
+    WebSocketAcceptEvent,
+    WebSocketCloseEvent,
+    WebSocketConnectEvent,
+    WebSocketDisconnectEvent,
+    WebSocketReceiveEvent,
+    WebSocketScope,
+    WebSocketSendEvent,
+)
```

Comment:
I'm actually a bit lukewarm on using `aspiref.typing` vs. just messages are annotated as `dict`.

There are some cases where using very precise typing feels like a port trade-off vs. readability and comprehensibility, and this case is one where I'm a bit "meh" about if it's a good trade-off or not.

I don't have any *absolute* feelings about that, but one option we *could* consider would be to type ASGI messages as `dict` as a first pass, and then have a look at a PR switch *all cases* over to `aspiref.typing` in one go.

As I say, I'm not absolute one way or the other, so if folks feel strongly here, then sure - let's go with the consensus. But just voicing that I sometimes find there are places where overly-typed annotations can start to become less comprehensible and useful than more loosely typed annotations.

Reply:
> I'm actually a bit lukewarm on using `aspiref.typing` vs. just messages are annotated as `dict`.

Opened at #1373